### PR TITLE
CORDA-3968: Prepare keystore handling for certificate rotation [OS]

### PR DESCRIPTION
**Pull request by rekalov**
Part of changes related to certificate re-issuance.

Ready for review. Do not merge yet - until ENT changes (https://github.com/corda/enterprise/pull/3823) are also reviwed. 

### Refactoring
- Move keystore validation logic from `AbstactNode` (and from `NodeKeyStoreUtilities`) to `KeyStoreHandler`.
- Combine initKeyStores() and obtainIdentities() into the single function (KeyStoreHandler.init())
- Extensive test coverage for keystore validation and key/certificate loading:
  - Use real *.jks files instead of mocks (previous version of tests in `NodeKeyStoreUtilitiesTest` using mocks didn't properly catch IO and password related problems).
  - Add more tests incl. notaries.
- Remove improper use of `AliasPrivateKey` (as a placeholder for key+alias, e.g. to pass signing keys + aliases to KeyManagementService). This is a preparation for further cleanup of `AliasPrivateKey`.

### Bug fixes
- Fix handling of absent keystore files and wrong passwords spotted by new tests (`getCertificateStores()`).

### Functional changes
- Stop generating node legal identity on the fly on the first node startup. Move it to node registration phase instead.
- Remove Node CA access at node normal runtime. No changes in registration phase (it's still generated).

### CODE REVIEW

**Review by thschroeter — COMMENTED** [review]
The notary parts look good to me.

File: node/src/main/kotlin/net/corda/node/internal/KeyStoreHandler.kt [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            }
+        }
+    }
+
+    private fun validateKeyStores(certStores: AllCertificateStores): X509Certificate {
+        // Check that trustStore contains the correct key-alias entry.
+        require(X509Utilities.CORDA_ROOT_CA in certStores.trustStore) {
+            "Alias for trustRoot key not found. Please ensure you have an updated trustStore file."
+        }
+        val trustRoot = certStores.trustStore[X509Utilities.CORDA_ROOT_CA]
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/internal/KeyStoreHandler.kt:96]: This seems to be assuming a single trust root, which seems retrograde

Comment (rekalov) [reply node/src/main/kotlin/net/corda/node/internal/KeyStoreHandler.kt:96]: This is intentional: I will add multi-root support in a separate PR.

**Review by mnesbit — APPROVED** [review]
OK so long as multi-root follows

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**